integration/test_collection_filter.py

```diff
@@ -0,0 +1,411 @@
+        )
+    )
+    now = datetime.datetime.now(datetime.timezone.utc)
+    later = now + datetime.timedelta(hours=1)
+    much_later = now + datetime.timedelta(days=1)
+
+    uuids = [
+        collection.data.insert({"dates": [now, later, much_later]}),
+        collection.data.insert({"dates": [now, now, much_later]}),
+        collection.data.insert({"dates": []}),
+        collection.data.insert({"dates": [much_later]}),
+    ]
+
+    if test_number == 1:
+        contains = [now, later]
+    else:
+        contains = [now]
+
+    weav_filter = Filter(path="dates").contains_any(contains)
+
+    objects = collection.query.get_flat(
+        filters=weav_filter, return_metadata=MetadataQuery(uuid=True)
+    )
+    assert len(objects) == len(results)
+
+    uuids = [uuids[result] for result in results]
+    assert all(obj.metadata.uuid in uuids for obj in objects)
+
+
+def test_ref_filters(client: weaviate.Client):
+    client.collection.delete("TestFilterRef")
+    client.collection.delete("TestFilterRef2")
+    to_collection = client.collection.create(
+        CollectionConfig(
+            name="TestFilterRef2",
+            vectorizer=Vectorizer.NONE,
+            properties=[Property(name="int", data_type=DataType.INT)],
+        )
+    )
+    uuid_to = to_collection.data.insert(data={"int": 0})
+    uuid_to2 = to_collection.data.insert(data={"int": 5})
+    from_collection = client.collection.create(
+        CollectionConfig(
+            name="TestFilterRef",
+            properties=[
+                ReferenceProperty(name="ref", target_collection="TestFilterRef2"),
+                Property(name="name", data_type=DataType.TEXT),
+            ],
+            vectorizer=Vectorizer.NONE,
+        )
+    )
+
+    from_collection.data.insert({"ref": ReferenceTo(uuid_to), "name": "first"})
+    from_collection.data.insert({"ref": ReferenceTo(uuid_to2), "name": "second"})
+
+    objects = from_collection.query.get_flat(
+        filters=Filter(path=["ref", "TestFilterRef2", "int"]).greater_than(3)
+    )
+    assert len(objects) == 1
+    assert objects[0].data["name"] == "second"
+
+
+def test_ref_filters_multi_target(client: weaviate.Client):
+    target = "TestFilterRefMulti2"
+    source = "TestFilterRefMulti"
+    client.collection.delete(source)
+    client.collection.delete(target)
+    to_collection = client.collection.create(
+        CollectionConfig(
+            name=target,
+            vectorizer=Vectorizer.NONE,
+            properties=[Property(name="int", data_type=DataType.INT)],
+        )
+    )
+    uuid_to = to_collection.data.insert(data={"int": 0})
+    uuid_to2 = to_collection.data.insert(data={"int": 5})
+    from_collection = client.collection.create(
+        CollectionConfig(
+            name=source,
+            properties=[
+                ReferencePropertyMultiTarget(
+                    name="ref", target_collections=[target, "TestFilterRefMulti"]
```

Comment:
Wow, collections can reference themselves?!

Reply:
yes :) Think of `Authors.cooperatesWith`